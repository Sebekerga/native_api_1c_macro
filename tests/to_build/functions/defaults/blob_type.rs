use std::sync::Arc;

use native_api_1c::native_api_1c_core::ffi::connection::Connection;
use native_api_1c_macro::AddIn;

#[derive(AddIn)]
pub struct MyAddIn {
    #[add_in_con]
    connection: Arc<Option<&'static Connection>>,

    #[add_in_func(name = "MyFunction", name_ru = "МояФункция")]
    #[arg(Blob, default = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9])]
    #[returns(Blob)]
    pub my_function: fn(&Self, Vec<u8>) -> Vec<u8>,
}

impl MyAddIn {
    pub fn new() -> Self {
        Self {
            connection: Arc::new(None),
            my_function: Self::my_function,
        }
    }

    fn my_function(&self, arg: Vec<u8>) -> Vec<u8> {
        arg
    }
}

fn main() {
    let _add_in = MyAddIn::new();
}
