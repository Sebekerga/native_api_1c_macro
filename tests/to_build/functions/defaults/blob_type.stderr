error: AddIn function attribute `arg` of type `Blob` cannot have default value
  --> tests/to_build/functions/defaults/blob_type.rs:12:6
   |
12 |     #[arg(Blob, default = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9])]
   |      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
